src/lib/containers/discussion_forum/DiscussionThread.tsx

```diff
@@ -35,15 +37,27 @@ export function DiscussionThread(props: DiscussionThreadProps) {
   const [showThreadModal, setShowThreadModal] = useState(false)
   const [showReplyEditor1, setShowReplyEditor1] = useState(false)
   const [showReplyEditor2, setShowReplyEditor2] = useState(false)
+  const [showDeleteModal, setShowDeleteModal] = useState(false)
 
   const { threadData, threadBody } = useGetThread(threadId)
   const { data: currentUserProfile } = useGetCurrentUserProfile()
```

Comment:
You should also be thinking about the anonymous user experience.  User might be able to READ the forum, but no user actions should be available (following, pinning, replying, ...)